src/utils/particle/repulser.ts

```diff
@@ -65,44 +71,48 @@ export class Repulser {
         }
     }
 
-    private hoverRepulse() {
+    private hoverRepulse(): void {
         const container = this.container;
         const options = container.options;
         const particle = this.particle;
-
-        let dx_mouse = particle.position.x - (container.interactivity.mouse.pos_x || 0);
-        let dy_mouse = particle.position.y - (container.interactivity.mouse.pos_y || 0);
-        let dist_mouse = Math.sqrt(dx_mouse * dx_mouse + dy_mouse * dy_mouse);
-        let normVec = { x: dx_mouse / dist_mouse, y: dy_mouse / dist_mouse };
-        let repulseRadius = options.interactivity.modes.repulse.distance, velocity = 100;
-        let repulseFactor = Utils.clamp((1 / repulseRadius) * (-1 * Math.pow(dist_mouse / repulseRadius, 2) + 1) * repulseRadius * velocity, 0, 50);
-        let pos = {
+        const dx_mouse = particle.position.x - (container.interactivity.mouse.pos_x || 0);
+        const dy_mouse = particle.position.y - (container.interactivity.mouse.pos_y || 0);
+        const dist_mouse = Math.sqrt(dx_mouse * dx_mouse + dy_mouse * dy_mouse);
+        const normVec = { x: dx_mouse / dist_mouse, y: dy_mouse / dist_mouse };
+        const repulseRadius = options.interactivity.modes.repulse.distance, velocity = 100;
+        const repulseFactor = Utils.clamp((1 - Math.pow(dist_mouse / repulseRadius, 2)) * velocity, 0, 50);
+        const pos = {
             x: particle.position.x + normVec.x * repulseFactor,
             y: particle.position.y + normVec.y * repulseFactor
         };
+        const outMode = options.particles.move.out_mode;
 
-        if (options.particles.move.out_mode === OutMode.bounce || options.particles.move.out_mode === OutMode.bounceVertical) {
-            if (pos.x - particle.radius > 0 && pos.x + particle.radius < container.canvas.w)
+        if (outMode === OutMode.bounce || outMode === OutMode.bounceVertical) {
+            if (pos.x - particle.radius > 0 && pos.x + particle.radius < container.canvas.w) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['magic numbers' are not allowed: 0](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5011230)